node/src/accountant/scanners/scan_schedulers.rs

```diff
@@ -0,0 +1,839 @@
+    fn resolve_rescheduling_for_given_error(
+        &self,
+        hintable_scanner: HintableScanner,
+        error: &StartScanError,
+        is_externally_triggered: bool,
+    ) -> SchedulingAfterHaltedScan {
+        match hintable_scanner {
+            HintableScanner::NewPayables => {
+                Self::resolve_new_payables(error, is_externally_triggered)
+            }
+            HintableScanner::RetryPayables => {
+                Self::resolve_retry_payables(error, is_externally_triggered)
+            }
+            HintableScanner::PendingPayables {
+                initial_pending_payable_scan,
+            } => Self::resolve_pending_payables(
+                error,
+                initial_pending_payable_scan,
+                is_externally_triggered,
+            ),
+        }
+    }
+}
+
+impl RescheduleScanOnErrorResolverReal {
+    fn resolve_new_payables(
+        err: &StartScanError,
+        is_externally_triggered: bool,
+    ) -> SchedulingAfterHaltedScan {
+        if is_externally_triggered {
+            SchedulingAfterHaltedScan::DoNotSchedule
+        } else if matches!(err, StartScanError::ScanAlreadyRunning { .. }) {
+            unreachable!(
+                "an automatic scan of NewPayableScanner should never interfere with itself {:?}",
+                err
+            )
+        } else {
+            SchedulingAfterHaltedScan::Schedule(ScanType::Payables)
+        }
+    }
+
+    // This looks paradoxical, but this scanner is meant to be shielded by the scanner right before
+    // it. That should ensure this scanner will not be requested if there was already something
+    // fishy. We can go strict.
+    fn resolve_retry_payables(
+        err: &StartScanError,
+        is_externally_triggered: bool,
+    ) -> SchedulingAfterHaltedScan {
+        if is_externally_triggered {
+            SchedulingAfterHaltedScan::DoNotSchedule
+        } else {
+            unreachable!(
+                "{:?} should be impossible with RetryPayableScanner in automatic mode",
+                err
+            )
+        }
+    }
+
+    fn resolve_pending_payables(
+        err: &StartScanError,
+        initial_pending_payable_scan: bool,
+        is_externally_triggered: bool,
+    ) -> SchedulingAfterHaltedScan {
+        if is_externally_triggered {
+            SchedulingAfterHaltedScan::DoNotSchedule
+        } else if err == &StartScanError::NothingToProcess {
+            if initial_pending_payable_scan {
+                SchedulingAfterHaltedScan::Schedule(ScanType::Payables)
+            } else {
+                unreachable!(
+                    "the automatic pending payable scan should always be requested only in need, \
+                    which contradicts the current StartScanError::NothingToProcess"
+                )
+            }
+        } else if err == &StartScanError::NoConsumingWalletFound {
+            SchedulingAfterHaltedScan::Schedule(ScanType::Payables)
```

Comment:
You'd like another check here which considers whether there is an awareness of unresolved pending payables.